arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -620,29 +619,31 @@ public void traceContextEnter(MessageFrame frame) {
                       .frame()
                       .getStackItem(currentOpCode.callMayNotTransferValue() ? 3 : 4));
 
-      final long callDataContextNumber = this.callStack.current().contextNumber();
+      final long callDataContextNumber = callStack.current().contextNumber();
 
-      this.callStack.enter(
-          this.state.stamps().hub(),
-          frame.getRecipientAddress(),
-          frame.getContractAddress(),
-          frame.getRecipientAddress(), // TODO: this is likely false
-          new Bytecode(frame.getCode().getBytes()),
+      // TODO: the
+      callStack.enter(
           frameType,
+          newChildContextNumber(),
+          this.deploymentStatusOf(frame.getContractAddress()),
           frame.getValue(),
           frame.getRemainingGas(),
+          frame.getRecipientAddress(),
+          this.deploymentNumberOf(frame.getRecipientAddress()),
+          frame.getContractAddress(),
+          this.deploymentNumberOf(frame.getContractAddress()),
+          new Bytecode(frame.getCode().getBytes()),
+          frame.getSenderAddress(), // TODO: is this better ?
           frame.getInputData(),
           callDataOffset,
           callDataSize,
-          callDataContextNumber,
-          this.transients.conflation().deploymentInfo().deploymentNumber(codeAddress),
-          codeDeploymentNumber,
-          isDeployment);
-      this.currentFrame().initializeFrame(frame); // TODO should be done in enter
```

Comment:
// TODO should be done in enter

because we do it both if depth == 0 and if depth > 0

Reply:
I tried doing it for once for both but it breaks even our tests. I think it's related to entering the modules.

Reply:
I tried doing it for once for both but it breaks even our tests. I think it's related to entering the modules.@